config-gen/README.md

```diff
@@ -0,0 +1,81 @@
+# Config-gen
+
+`kubebuilder alpha config-gen` is a subcommand that generates configuration for kubebuilder projects as a configuration function.
+
+Supports:
+
+- Generating CRDs and RBAC from code
+- Generating webhook certificates for development
+- Selectively enabling / disabling components such as prometheus and webhooks
+  - See [types.go](apis/v1alpha1/types.go) for a list of components
+
+## Usage
+
+`config-gen` may be run as a standalone command or from kustomize as a transformer plugin.
+
+### Standalone command
+
+config-gen may be run as a standalone program on the commandline.
+
+See [examples/standalone](examples/standalone/README.md)
+
+### From kustomize
+
+config-gen may be run as a Kustomize plugin using kustomize.
+
+See [examples/kustomize](examples/kustomize/README.md)
+
+### Extending `config-gen`
+
+config-gen may be extended by composing additional functions on top of it.
+
+See examples of layering additional functions on:
+
+- [examples/basicextension](examples/basicextension/README.md)
+- [examples/advancedextension](examples/advancedextension/README.md)
+
+## `KubebuilderProject`
+
+See [types.go](apis/v1alpha1/types.go) for KubebuilderProject schema.
+
+Following is an example of a `kubebuilderproject.yaml` used to generate configuration.
+
+```yaml
+apiVersion: kubebuilder.sigs.k8s.io
+kind: KubebuilderProject
+metadata:
+  name: simple
+spec:
+  # container image containing the controller-manager
+  image: example/simple:latest
+
+  # kubebuilder project root directory relative path from kubebuilderproject.yaml dir
+  # (defaults to current directory)
+  projectDirectory: ../project/...
+
+  # components to enable / disable generating config for
+#  components:
+#    namespace: false
+#    controller-manager: false
+#    crds: false
+#    rbac: false
+#    cert-manager: true
+#    prometheus: true
+
+   # enable conversion webhooks for these groups
+#  conversionWebhooks:
+#    "bars.example.my.domain": true
+
+   # disable the auth proxy
+#  disableAuthProxy: true
+
+   # use component config
+#  componentConfigFilepath: ./relative/path/to/controller_manager_config.yaml
+
+  # generate certificates for webhooks
+#  developmentOptions:
+#    generateCert: true
+#    certDuration: '24h'
+```
```

Comment:
@pwittrock could we add an option here to enable write on disk as an option before we merge it such as alpha?

Reply:
What is the use case where letting the shell handle this is insufficient?

Reply:
- I am as an operator developer would like to have the config/ directory scaffolded by config-gen so that, I do not need to work with kustomize patches and I can change the manually. 
- I am as kb/sdk maintainer would like to replace the config/ directory by config-gen without losing the backwards compatibility. 
 
So, my idea was we have an option for that which by default would be false. However, we can do that in follow-ups as well.

Reply:
Ah that makes sense.  Lets get the command stable first, then figure out a migration path.